src/snowflake/cli/plugins/spcs/image_registry/commands.py

```diff
@@ -19,5 +23,19 @@
 @with_output
 @global_options_with_connection
 def token(**options) -> ObjectResult:
-    """Gets the token from environment to use for authenticating with the registry."""
+    """Gets the token from environment to use for authenticating with the registry. Note that this token is specific
+    to your current user and will not grant access to any repositories that your current user cannot access."""
     return ObjectResult(RegistryManager().get_token())
+
+
+@app.command()
+@with_output
+@global_options_with_connection
+def url(**options) -> MessageResult:
+    """Gets the image registry URL for the current account. Must be called from a role that can view at least one image repository in the image registry."""
+    try:
+        return MessageResult(RegistryManager().get_registry_url())
+    except NoRepositoriesViewableError:
+        raise ClickException(
+            "Current role cannot view any image repositories. Please ensure that at least one repository exists in your image registry and use a role with read access to at least one image repository before retrieving registry URL."
```

Comment:
"cannot view" sounds like the role is prohibited to view any repository. Could we rephrase it to sth like
`No image repository found. To get the registry url, please switch to a role with read access to at least one image repository or create a new image repository first`